app/src/lib/git/diff.ts

```diff
@@ -365,6 +362,52 @@ function diffFromRawDiffOutput(output: Buffer): IRawDiff {
   return parser.parse(pieces[pieces.length - 1])
 }
 
+/**
+ * Utility function that takes a buffer and outputs
+ * either a largeTextDiff or null
+ */
+function buildLargeTextDiff(
+  buffer: Buffer,
+  lineEndingsChange?: LineEndingsChange
+): ILargeTextDiff | null {
+  if (!isValidBuffer(buffer)) {
+    // we know we can't transform this process output into a diff, so let's
+    // just return a placeholder for now that we can display to the user
+    // to say we're at the limits of the runtime
+    return { kind: DiffType.LargeText, length: buffer.length }
+  }
+
+  const diffText = diffFromRawDiffOutput(buffer)
+
+  if (isBufferTooLarge(buffer)) {
+    // we don't want to render by default
+    // but we keep it as an option by
+    // passing in text and hunks
+    const largeTextDiff: ILargeTextDiff = {
+      kind: DiffType.LargeText,
+      length: buffer.length,
```

Comment:
if we remove the length in the helper text in the component, we don't need this any more and we can combine this with the one below, right? am i missing where it's being used?